.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,22 @@
+name: Pull request checks
+
+on: [pull_request]
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
Why not extract shellchecking into its own workflow script that gets run under all of PR submissions, changed tags, and workflow_dispatch?

Reply:
What is the benefit of this? The current `release.yaml` workflow will not create a release if the `shellcheck` step fails. This would be harder to do if shellchecking is in a separate workflow.